### PR TITLE
Allow to configure github and bitbucket oauth config

### PR DESCRIPTION
**Pull request by tolusha**
Signed-off-by: Anatolii Bazko <abazko@redhat.com>

### What does this PR do?
Allow to configure github and bitbucket oauth config

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/18932

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [Custom resource definition file is up to date](https://github.com/eclipse/che-operator/blob/master/README.md#updating-custom-resource-definition-file)
- [x] [Nightly OLM bundle is up to date](https://github.com/eclipse/che-operator/blob/master/README.md#update-nightly-olm-bundle)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: pkg/deploy/sercet_test.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+	"testing"
+)
+
+func TestGetSecrets(t *testing.T) {
+	type testCase struct {
+		name          string
+		labels        map[string]string
+		annotations   map[string]string
+		initObjects   []runtime.Object
+		exectedAmount int
```
Comment (AndrienkoAleksandr) [inline pkg/deploy/sercet_test.go:35]:
```suggestion
		expectedAmount int
```

File: pkg/deploy/server/deployment_che.go [hunk 2]
```diff
@@ -406,3 +411,48 @@ func GetFullCheServerImageLink(checluster *orgv1.CheCluster) string {
...
+		deployment.Spec.Template.Spec.Volumes = append(deployment.Spec.Template.Spec.Volumes,
+			corev1.Volume{
+				Name: secrets[0].Name,
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{
+						SecretName: secrets[0].Name,
+					},
+				},
+			})
+		deployment.Spec.Template.Spec.Containers[0].VolumeMounts = append(deployment.Spec.Template.Spec.Containers[0].VolumeMounts,
```
Comment (AndrienkoAleksandr) [inline pkg/deploy/server/deployment_che.go:437]: We can move deployment.Spec.Template.Spec.Containers[0] to the local variable with name 'container' or something like that.

**Review by AndrienkoAleksandr — APPROVED** [review]
Maybe we should at least log warning if we detected more than one secret...

**Comment by tolusha** [comment]
The problem with a message that it will be printed on every reconcile loop.

File: pkg/deploy/server/deployment_che.go (hunk 2 above)
Comment (tolusha) [reply pkg/deploy/server/deployment_che.go:437]: `container := &deployment.Spec.Template.Spec.Containers[0]`
sounds good

**Review by AndrienkoAleksandr — APPROVED** [review]

**Pull request merged**